test/cdp-proxy/reactNativeCDPProxy.test.ts

```diff
@@ -0,0 +1,318 @@
+    const cdpProxyPort = generateRandomPortNumber();
+    const cdpProxyLogLevel = LogLevel.Custom;
+    const proxy = new ReactNativeCDPProxy(cdpProxyHostAddress, cdpProxyPort);
+
+    const wsTargetPort = generateRandomPortNumber();
+    let wsTargetServer: Server | null;
+
+    let targetConnection: Connection | null;
+    let debugConnection: Connection | null;
+
+    suiteSetup(async () => {
+      proxy.setApplicationTargetPort(wsTargetPort);
+      await proxy.initializeServer(new RnCDPMessageHandler(), cdpProxyLogLevel);
+
+      await Server.create({ host: "localhost", port: wsTargetPort })
+        .then((server: Server) => {
+          wsTargetServer = server;
+
+          server.onConnection(([connection, request]: [Connection, Request]) => {
+            targetConnection = connection;
+          });
+        });
+
+      const proxyUri = await new DebuggerEndpointHelper().getWSEndpoint(`http://${cdpProxyHostAddress}:${cdpProxyPort}`);
+      debugConnection = new Connection(await WebSocketTransport.create(proxyUri));
+
+      while (!targetConnection) {
+        await promiseUtil.delay(1000);
+      }
+    });
+
+    suiteTeardown(() => {
+      if (targetConnection) {
+        targetConnection.close();
+        targetConnection = null;
+      }
+      if (debugConnection) {
+        debugConnection.close();
+        debugConnection = null;
+      }
+      proxy.stopServer();
+      if (wsTargetServer) {
+        wsTargetServer.dispose();
+        wsTargetServer = null;
+      }
+    });
+
+    suite("MessageHandlers", () => {
+      let rnHandler = new RnCDPMessageHandler();
+      let directHandler = new DirectCDPMessageHandler();
+
+      const deliverlyTest = function (messageHandler: ICDPMessageHandler) {
+
+        return test(`Messages should be delivered correctly with ${(<Object>messageHandler).constructor.name}`, async () => {
+          const targetMessageStart = {method: "Target.start", params: {reason: "test"}};
+          const debuggerMessageStart = {method: "Debugger.start", params: {reason: "test"}};
+
+          const messageFromTarget = await new Promise((resolve) => {
+            targetConnection?.send(targetMessageStart);
+
+            debugConnection?.onCommand((evt: IProtocolCommand) => {
+              resolve(evt);
+            });
+          })
+           .then((evt) => {
+            return evt;
+          });
+
+          const messageFromDebugger = await new Promise((resolve) => {
+            debugConnection?.send(debuggerMessageStart);
+
+            targetConnection?.onCommand((evt: IProtocolCommand) => {
+              resolve(evt);
+            });
+          })
+           .then((evt) => {
+            return evt;
+          });
+
+          assert.deepEqual(messageFromTarget, targetMessageStart);
+          assert.deepEqual(messageFromDebugger, debuggerMessageStart);
+        });
+      };
+
+      suite(`${(<Object>rnHandler).constructor.name}`, () => {
```

Comment:
Do we need this type cast here and further? It seems that all works fine even without this cast.